versioning.mk

```diff
@@ -0,0 +1,50 @@
+VERSION_PREFIX := v
+# check if git is available
+ifeq ($(shell which git),)
+        $(warning git is not available, binaries will not include git SHA)
+        GIT_SHA :=
+        GIT_TREE_STATE :=
+        GIT_TAG :=
+        VERSION_SUFFIX := unknown
+else
+        GIT_SHA := $(shell git rev-parse --short HEAD)
+        # Tree state is "dirty" if there are uncommitted changes, untracked files are ignored
+        GIT_TREE_STATE := $(shell test -n "`git status --porcelain --untracked-files=no`" && echo "dirty" || echo "clean")
+        # Empty string if we are not building a tag
+        GIT_TAG := $(shell git describe --tags --abbrev=0 --exact-match 2>/dev/null)
+        ifeq ($(GIT_TREE_STATE),dirty)
+                VERSION_SUFFIX := $(GIT_SHA).dirty
+        else
+                VERSION_SUFFIX := $(GIT_SHA)
+        endif
+endif
+
+# if building a tag or VERSION is set, set RELEASE_STATUS to "released"
+ifdef VERSION
+        RELEASE_STATUS := released
+else ifneq ($(GIT_TAG),)
+        RELEASE_STATUS := released
+else
+        RELEASE_STATUS := unreleased
+endif
+
+ifndef VERSION
+        VERSION := $(shell head -n 1 VERSION)
+        DOCKER_IMG_VERSION := $(VERSION_PREFIX)$(VERSION)-$(VERSION_SUFFIX)
+else
+        DOCKER_IMG_VERSION := $(VERSION)
+endif
+
+# Default bundle image tag
+BUNDLE_IMG ?= antrea/antrea-operator-bundle:$(DOCKER_IMG_VERSION)
+
+# Image URL to use all building/pushing image targets
+IMG ?= antrea/antrea-operator:$(DOCKER_IMG_VERSION)
```

Comment:
nit: can we call this variable OPERATOR_IMG?

Reply:
Done.